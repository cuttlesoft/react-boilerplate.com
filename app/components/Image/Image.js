/* eslint-disable react/prop-types */
/**
 *
 * Image
 *
 */

import React from 'react'
import { Image } from 'grommet'

function createImage({ a11yTitle, alignSelf, fallback, fit, gridArea, margin, opacity, src }) {
  return (
    <Image
      a11yTitle={a11yTitle}
      alignSelf={alignSelf}
      fallback={fallback}
      fit={fit}
      gridArea={gridArea}
      margin={margin}
      opacity={opacity}
      src={src}
    />
  )
}

let createImageWithDoc

if (process.env.NODE_ENV !== 'production') {
  createImageWithDoc = require('./Image.doc').doc(createImage) // eslint-disable-line global-require
}

const ImageWrapper = createImageWithDoc || createImage
export default ImageWrapper
