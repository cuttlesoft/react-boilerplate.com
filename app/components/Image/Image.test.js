import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Image from './Image'

const renderComponent = (props = {}) => render(<Image {...props} />)

/**
 *
 * Tests for Image
 *
 */
describe('Image', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
