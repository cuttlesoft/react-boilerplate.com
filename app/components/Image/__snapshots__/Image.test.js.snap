// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders and matches snapshot 1`] = `
<div>
  <img
    class=""
  />
</div>
`;
